tests/integration/telemetry/stats/prometheus/footprint/footprint_test.go

```diff
@@ -0,0 +1,474 @@
+		"istio_request_bytes_count":                                                    true,
+		"istio_request_bytes_sum":                                                      true,
+		"istio_request_duration_milliseconds_bucket":                                   true,
+		"istio_request_duration_milliseconds_count":                                    true,
+		"istio_request_duration_milliseconds_sum":                                      true,
+		"istio_response_bytes_bucket":                                                  true,
+		"istio_response_bytes_count":                                                   true,
+		"istio_response_bytes_sum":                                                     true,
+	}
+)
+
+func TestExpectedStats(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.stats.prometheus.http.nullvm").
+		Run(func(ctx framework.TestContext) {
+			g, _ := errgroup.WithContext(context.Background())
+			for _, cltInstance := range common.GetClientInstances() {
+				cltInstance := cltInstance
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := common.SendTraffic(cltInstance); err != nil {
+							return err
+						}
+						return nil
+					}, retry.Delay(framework.TelemetryRetryDelay), retry.Timeout(framework.TelemetryRetryTimeout))
+					if err != nil {
+						return err
+					}
+					return nil
+				})
+			}
+			if err := g.Wait(); err != nil {
+				t.Fatalf("test failed: %v", err)
+			}
+			retry.UntilSuccessOrFail(t, func() error {
+				prom := common.GetPromInstance()
+				lv, err := prom.KnownMetrics()
+				if err != nil {
+					t.Fatalf("could not retrieve list of known metrics: %v", err)
+				}
+				t.Logf("known metrics retrieved: %#v", lv)
+				metricsMap := map[string]bool{}
+				for _, lval := range lv {
+					str := string(lval)
+					if strings.HasPrefix(str, "istio_echo") {
+						// ignore the echo app metrics
+						continue
+					}
+					if strings.HasPrefix(str, "envoy_cluster_zone_") {
+						// ignore the zone-specific envoy metrics, as the tests could be run in any location
+						// todo(douglas-reid): decide how to "fix" the broken zone labels
+						continue
+					}
+					if strings.HasPrefix(str, "istio_") || strings.HasPrefix(str, "envoy_") {
+						metricsMap[string(lval)] = true
+					}
+				}
+
+				if diff := cmp.Diff(canonicalMetrics, metricsMap); diff != "" {
+					return fmt.Errorf("retrieved different set of metrics: %v", diff)
+				}
+				return nil
+			}, retry.Delay(10*time.Second), retry.Timeout(2*time.Minute))
+
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.NewSuite(m).
+		Label(label.CustomSetup).
+		Label(label.IPv4). // https://github.com/istio/istio/issues/35915
+		Setup(istio.Setup(common.GetIstioInstance(), setupConfig)).
+		Setup(common.TestSetup).
+		Run()
+}
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	// enable telemetry v2 with nullvm
```

Comment:
do we need a new test suite? Or could we use the existing nullvm one?

Reply:
Probably don't need a full test suite. It was cleaner/easier to separate at first, but I can collapse it.

Reply:
moved it.